osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetAllAttributeValuesCommandExecutorTest.java

```diff
@@ -0,0 +1,325 @@
+        new GetResultImpl(objectList, AccessResultCode.SUCCESS);
+
+    when(this.connectionManager.getConnection().get(any(AttributeAddress.class)))
+        .thenReturn(getResultObjectList);
+
+    when(this.connectionManager.getConnection().get(ArgumentMatchers.anyList()))
+        .thenReturn(this.createGetResultsForAttributes(KNOWN_OBIS, 25, 1, 30))
+        .thenReturn(this.createGetResultsForAttributes(UNKNOWN_OBIS, 26, null, null));
+
+    final String result = this.executor.execute(this.connectionManager, DEVICE, null, MSG_METADATA);
+
+    assertThat(removeNewLines(result))
+        .isEqualToIgnoringNewLines(
+            """
+[ {
+  "description" : "Active energy import (+A)",
+  "dlmsClass" : "REGISTER",
+  "version" : 0,
+  "obis" : "1.0.1.8.0.255",
+  "attributes" : [ {
+    "id" : 1,
+    "description" : "logical_name",
+    "rawValue" : "OCTET_STRING, ByteArray, value=[B]: logical name: 1-0:1.8.0.255",
+    "value" : "1.0.1.8.0.255",
+    "access" : "R"
+  }, {
+    "id" : 2,
+    "description" : "value",
+    "datatype" : "double-long-unsigned",
+    "rawValue" : "LONG_UNSIGNED, Number, value=[Integer]: 25",
+    "value" : "25",
+    "access" : "W"
+  }, {
+    "id" : 3,
+    "description" : "scaler_unit",
+    "datatype" : "scal_unit_type",
+    "rawValue" : "STRUCTURE, Complex, value=[Arrays$ArrayList]: [\\tLONG_UNSIGNED, Number, value=[Integer]: 1\\tLONG_UNSIGNED, Number, value=[Integer]: 30]",
+    "value" : "1, WH",
+    "access" : "RW"
+  } ],
+  "class-id" : 3
+}, {
+  "dlmsClass" : "DATA",
+  "version" : 1,
+  "obis" : "1.1.1.1.1.255",
+  "attributes" : [ {
+    "id" : 1,
+    "description" : "logical_name",
+    "rawValue" : "OCTET_STRING, ByteArray, value=[B]: logical name: 1-1:1.1.1.255",
+    "value" : "1.1.1.1.1.255",
+    "access" : "R"
+  }, {
+    "id" : 2,
+    "description" : "value",
+    "rawValue" : "LONG_UNSIGNED, Number, value=[Integer]: 26",
+    "value" : "26",
+    "access" : "W"
+  } ],
+  "class-id" : 1
+} ]
+""");
+  }
+
+  @Test
+  void testDecodingFailsForObject() throws Exception {
+    final DataObject objListElementForFailingObject =
+        this.createObjectListElement(CLASS_ID_DATA, VERSION_0, KNOWN_OBIS, NO_OF_ATTR_DATA);
+    final DataObject objectList = DataObject.newArrayData(List.of(objListElementForFailingObject));
+    final GetResultImpl getResultObjectList =
+        new GetResultImpl(objectList, AccessResultCode.SUCCESS);
+
+    when(this.connectionManager.getConnection().get(any(AttributeAddress.class)))
+        .thenReturn(getResultObjectList);
+
+    // Object of class 1 has unexpectedly more attributes, causing a decoding failure
+    when(this.connectionManager.getConnection().get(ArgumentMatchers.anyList()))
+        .thenReturn(this.createGetResultsForAttributes(KNOWN_OBIS, 26, 2, 31));
+
+    final String result = this.executor.execute(this.connectionManager, DEVICE, null, MSG_METADATA);
+
+    assertThat(removeNewLines(result))
+        .isEqualToIgnoringNewLines(
```

Comment:
Is it necessairy to both remove and ignore new lines?

Reply:
Running tests locally or on Jenkins produced different new lines characters, which aren't all ignored by the `isEqualToIgnoringNewLines` function. Function `removeNewLines` removes all new lines.

The resulting json is hard to read. To keep the test readable, the json in the test is with new lines. Therefore we still need the `isEqualToIgnoringNewLines`.